server/src/internalClusterTest/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java

```diff
@@ -158,7 +161,7 @@ protected Settings nodeSettings(int nodeOrdinal) {
 
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Collections.singletonList(InternalSettingsPlugin.class);
+        return Stream.of(InternalSettingsPlugin.class, MockInternalClusterInfoService.TestPlugin.class).collect(Collectors.toList());
```

Comment:
Maybe use `List.of(InternalSettingsPlugin.class, MockInternalClusterInfoService.TestPlugin.class)` instead to be a bit more concise?

Reply:
Addressed the comment.

> > > Another issue observed was there was race condition due to available disk space not getting refreshed frequently. Reducing update interval for `ClusterInfoService` and retrying a couple of more times.
> > 
> > 
> > After reducing the refresh frequency, retried 6 times. Did not observed `DiskThresholdDeciderIT` test cases failing in any of these runs.
> 
> Is there any way to make this deterministic? Reducing the refresh frequency may have made the race condition less likely, but has it actually been fixed?

Made few portion of test cases synchronous to make these test cases deterministic. Tested it out with 50 iterations run with no failure.

Reply:
Addressed the comment